api/src/legacy_soap_api/grantors/services/get_submission_list_expanded_response.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from datetime import datetime
+
+from sqlalchemy import select
+from sqlalchemy.orm import aliased, selectinload
+
+import src.adapters.db as db
+from src.constants.lookup_constants import ApplicationStatus
+from src.db.models.competition_models import Application, ApplicationSubmission, Competition
+from src.db.models.opportunity_models import Opportunity, OpportunityAssistanceListing
+from src.legacy_soap_api.grantors import schemas
+from src.legacy_soap_api.legacy_soap_api_utils import convert_bool_to_yes_no
+from src.util.datetime_util import adjust_timezone
+
+logger = logging.getLogger(__name__)
+
+GRANTS_APPLICATION_STATUSES = {
+    None: None,
+    ApplicationStatus.IN_PROGRESS: None,
+    ApplicationStatus.SUBMITTED: "Received",
+    ApplicationStatus.ACCEPTED: "Validated",
+}
+
+
+def transform_submission(submission: ApplicationSubmission) -> dict[str, str | datetime | None]:
+    application = submission.application
+    competition = application.competition
+    opportunity = competition.opportunity
+    assistance_listing = competition.opportunity_assistance_listing
+    application_list_obj = {
+        "FundingOpportunityNumber": opportunity.opportunity_number,
+        "CFDANumber": assistance_listing.assistance_listing_number if assistance_listing else None,
+        "GrantsGovTrackingNumber": f"GRANT{submission.legacy_tracking_number}",
+        "GrantsGovApplicationStatus": GRANTS_APPLICATION_STATUSES[application.application_status],
+        "SubmissionTitle": application.application_name,
+        "PackageID": competition.legacy_package_id,
+        "n2:ReceivedDateTime": (
+            adjust_timezone(application.submitted_at, "America/New_York")
+            if application.submitted_at
+            else None
+        ),
+        "SubmissionMethod": "web",
+        "DelinquentFederalDebt": None,
+        "ActiveExclusions": None,
+        "UEI": None,
+    }
+    if organization := application.organization:
+        sam_gov_entity = organization.sam_gov_entity
+        application_list_obj.update(
+            {
+                "DelinquentFederalDebt": (
+                    convert_bool_to_yes_no(
+                        sam_gov_entity is not None and sam_gov_entity.has_debt_subject_to_offset
+                    )
+                ),
+                "ActiveExclusions": (
+                    convert_bool_to_yes_no(
+                        sam_gov_entity is not None and sam_gov_entity.has_exclusion_status
+                    )
+                ),
+                "UEI": sam_gov_entity.uei if sam_gov_entity else None,
+            }
+        )
```

Comment:
Since the sam gov entity is really what you want, don't check if it's null in each separate place, just don't do anything if it's null.

```suggestion
    # something roughly like this syntax, not certain this is exactly valid
    if application.organization and sam_gov_entity := application.organization.sam_gov_entity:
        application_list_obj["DelinquentFederalDebt"] = convert_bool_to_yes_no(sam_gov_entity.has_debt_subject_to_offset)
        application_list_obj["ActiveExclusions"] = convert_bool_to_yes_no(sam_gov_entity.has_exclusion_status)
        application_list_obj["UEI"] = sam_gov_entity.uei 

```